accounts-journey/presentation/viewmodel/AccountUIMapper.swift

```diff
@@ -0,0 +1,257 @@
+    func mapToUi() -> AccountsUiModel {
+        
+        var accountsUiModel = AccountsUiModel()
+        if !self.products.isEmpty {
+            let config: AccountsJourney.Configuration = Resolver.resolve()
+            accountsUiModel = AccountsUiModel(
+                header: config.strings.generalAccountTitle(),
+                products: self.products.map {
+                    AccountUiModel(
+                        id: $0.identifier,
+                        name: $0.name,
+                        balance: Currency(
+                            amount: $0.bookedBalance,
+                            currencyCode: $0.currency
+                        ),
+                        state: formatState($0.state, number: $0.iban),
+                        isVisible: $0.visible,
+                        iconName: config.design.accountIcon(.general)
+                    )
+                })
+        }
+        return accountsUiModel
+    }
+}
+
+extension AccountsJourney.CurrentAccounts {
+    func mapToUi() -> AccountsUiModel? {
+        if !self.products.isEmpty {
+            let config: AccountsJourney.Configuration = Resolver.resolve()
+            return AccountsUiModel(
+                header: config.strings.currentAccountTitle(),
+                products: self.products.map {
+                    return AccountUiModel(
+                        id: $0.identifier,
+                        name: $0.name,
+                        balance: Currency(
+                            amount: $0.bookedBalance,
+                            currencyCode: $0.currency
+                        ),
+                        state: formatState($0.state, number: $0.bban ?? $0.iban ?? $0.bic),
+                        isVisible: $0.visible,
+                        iconName: config.design.accountIcon(.current)
+                    )
+                })
+        } else {
+            return nil
+        }
+    }
+}
+
+extension AccountsJourney.SavingsAccounts {
+    func mapToUi() -> AccountsUiModel? {
+        if !self.products.isEmpty {
+            let config: AccountsJourney.Configuration = Resolver.resolve()
+            return AccountsUiModel(
+                header: config.strings.savingsAccountTitle(),
+                products: self.products.map {
+                    return AccountUiModel(
+                        id: $0.identifier,
+                        name: $0.name,
+                        balance: Currency(
+                            amount: $0.bookedBalance,
+                            currencyCode: $0.currency
+                        ),
+                        state: formatState($0.state, number: $0.bban ?? $0.iban ?? $0.bic),
+                        isVisible: $0.visible,
+                        iconName: config.design.accountIcon(.savings)
+                    )
+                })
+        } else {
+            return nil
+        }
+    }
+}
+
+extension AccountsJourney.TermDeposits {
+    func mapToUi() -> AccountsUiModel? {
```

Comment:
I would remove all of these if we are simplifying the journey and just keep two as an example, because it's just code duplication and it doesn't help understanding, but will create a ticket for it and we can check it together in refinement.

Reply:
Definitely in agreement, for now we'll keep it because of alignment with Android